url_util.cc

```diff
@@ -279,6 +281,84 @@ bool DoCanonicalize(const CHAR* spec,
   return success;
 }
 
+template <typename CHAR>
```

Comment:
Is this bit of code (up to line 361) copied from somewhere?

Reply:
Yeah it's from the `DoCanonicalize` function. However, I don't want to change the `DoCanonicalize` because it has a lot to do with other parts of the GURL code so I made a copy of it instead and make some changes there 😄